### PR TITLE
Merge 23.6 final to trunk

### PR DESCRIPTION
**Pull request by oguzkocer**
There was a merge conflict in `WordPress/src/main/java/org/wordpress/android/ui/posts/PostListMainViewModel.kt`:

```
<<<<<<< HEAD
        site.isWPCom && site.hasCapabilityEditOthersPosts && (site.isSingleUserSite != null && !site.isSingleUserSite)
=======
        site.isUsingWpComRestApi && site.hasCapabilityEditOthersPosts
                && (site.isSingleUserSite != null && !site.isSingleUserSite)
>>>>>>> origin/trunk
```

I've resolved the conflict by selecting the implementation from `trunk`. @zwarm could you please verify that this is the correct resolution?

### CODE REVIEW

**Review by mokagio — APPROVED** [review]
- [x] `versionName` dropping the `rc` suffix and `versionCode` bump in `version.properties`
- [x] `strings.xml` updates (none since rc-1)
- [x] `release_notes.xml` and `changelogs/*.txt` updates
- [x] Diff from PRs that made it into the `release/` branch since the last beta 

Approving to unblock, but I disabled auto-merge to wait for the conflict resolution confirmation.

**Comment by oguzkocer** [comment]
Unfortunately I need to merge this PR without the requested confirmation to start the `23.7` code freeze.

**Pull request merged**
